### PR TITLE
Datablocks bid adapter: update adapter to conform to new bid server's format

### PR DESCRIPTION
**Pull request by htang555**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

Updated the bid adapter to conform to our new bid server's format. Also added a new optional parameter to the request to specify placement.

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [x] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- https://github.com/prebid/prebid.github.io/pull/2931

### CODE REVIEW

**Comment by aleksatr** [comment]
Hi @htang555,

- fix datablocksBidAdapter.md file, errors introduced in #4195.
- specify `usp_supported` in pbjs docs repo PR. 
- you specified that `host` param is required in your pbjs docs, but it looks optional, judging by your code. 
- fpd to be replaced with ortb2, check: https://docs.prebid.org/features/firstPartyData.html
- remove blocks // TODO: THIS IS TEMP WHILE TESTING

ESLint reports a bunch of errors when I run `gulp test`
```
$ gulp test

Prebid.js/modules/datablocksBidAdapter.js
    4:1   error  Expected consistent spacing                       standard/object-curly-even-spacing
  124:9   error  Identifier 'db_id' is not in camel case           camelcase
  126:7   error  Identifier 'db_id' is not in camel case           camelcase
  129:10  error  Identifier 'db_id' is not in camel case           camelcase
  130:7   error  Identifier 'db_id' is not in camel case           camelcase
  132:12  error  Identifier 'db_id' is not in camel case           camelcase
  138:11  error  Identifier 'sync_obj' is not in camel case        camelcase
  144:11  error  Identifier 'stored_syncs' is not in camel case    camelcase
  154:11  error  Identifier 'sync_data' is not in camel case       camelcase
  155:11  error  Identifier 'sync_data' is not in camel case       camelcase
  201:9   error  Identifier 'bot_test' is not in camel case        camelcase
  254:27  error  Identifier 'valid_requests' is not in camel case  camelcase
  254:43  error  Identifier 'bidder_request' is not in camel case  camelcase
  302:28  error  Identifier 'bid_request' is not in camel case     camelcase
  390:9   error  Identifier 'source_id' is not in camel case       camelcase
  395:2   error  Unexpected tab character                          no-tabs
  395:2   error  Mixed spaces and tabs                             no-mixed-spaces-and-tabs
  396:2   error  Unexpected tab character                          no-tabs
  396:2   error  Mixed spaces and tabs                             no-mixed-spaces-and-tabs
  396:23  error  Identifier 'source_id' is not in camel case       camelcase
  396:57  error  Identifier 'source_id' is not in camel case       camelcase
  398:2   error  Unexpected tab character                          no-tabs
  398:2   error  Mixed spaces and tabs                             no-mixed-spaces-and-tabs
  411:35  error  Identifier 'rtb_response' is not in camel case    camelcase
  411:49  error  Identifier 'gdpr_consent' is not in camel case    camelcase
  411:63  error  Identifier 'usp_consent' is not in camel case     camelcase
  413:9   error  Identifier 'bid_response' is not in camel case    camelcase
  427:9   error  Identifier 'gdpr_data' is not in camel case       camelcase
  431:16  error  Identifier 'gdpr_consent' is not in camel case    camelcase
  441:9   error  Identifier 'db_buyer_id' is not in camel case     camelcase
  443:7   error  Identifier 'db_buyer_id' is not in camel case     camelcase
  457:14  error  Identifier 'add_params' is not in camel case      camelcase
  460:11  error  Identifier 'url_params' is not in camel case      camelcase
  467:24  error  Identifier 'db_buyer_id' is not in camel case     camelcase
  477:14  error  Identifier 'check_valid' is not in camel case     camelcase
  484:11  error  Unreachable code                                  no-unreachable
  487:11  error  Unreachable code                                  no-unreachable
  490:11  error  Unreachable code                                  no-unreachable
  508:31  error  Identifier 'rtb_response' is not in camel case    camelcase
  508:45  error  Identifier 'bid_request' is not in camel case     camelcase
  543:9   error  Unexpected tab character                          no-tabs
  548:9   error  Identifier 'res_bids' is not in camel case        camelcase
  554:11  error  Identifier 'result_item' is not in camel case     camelcase
  556:11  error  Identifier 'media_type' is not in camel case      camelcase
  557:15  error  Identifier 'media_type' is not in camel case      camelcase
  563:15  error  Identifier 'native_result' is not in camel case   camelcase
  637:3   error  Identifier 'do_tests' is not in camel case        camelcase
  639:17  error  'test' is assigned a value but never used         no-unused-vars

✖ 48 problems (48 errors, 0 warnings)

[12:30:59] 'lint' errored after 34 s
[12:30:59] ESLintError in plugin "gulp-eslint"
```

File: modules/datablocksBidAdapter.js [hunk 1]
```diff
@@ -1,330 +1,660 @@
...
-    return !!(bid.params.host && bid.params.sourceId &&
-      bid.mediaTypes && (bid.mediaTypes.banner || bid.mediaTypes.native || bid.mediaTypes.video));
-  },
-  buildRequests: function(validBidRequests, bidderRequest) {
-    if (!validBidRequests.length) { return []; }
+	supportedMediaTypes: [BANNER, NATIVE],
+	code: 'datablocks',
+	
+	// DATABLOCKS SCOPED OBJECT
+	db_obj: {metrics_host: 'prebid.datablocks.net', metrics: [], metrics_timer: null, metrics_queue_time: 1000, vis_optout: false, source_id: 0},
```
Comment (aleksatr) [inline modules/datablocksBidAdapter.js:99]: @htang555 isn't this better suited for your analytics adapter?

Comment (htang555) [reply modules/datablocksBidAdapter.js]: This is for preliminary viewability metrics that are required for our adapter to perform (there is an option for the user to opt out of this in the bid params).

**Comment by htang555** [comment]
I updated the prebid.github.io PR with usp_supported and the bid param changes.

**Comment by aleksatr** [comment]
@htang555 Video example in datablocksBidAdapter.md is misplaced inside bids array of the native ad unit example. From what I see you're removing video support from this adapter, so you should probably remove the video example, instead of fixing it.

**Comment by jmayor** [comment]
Updated the readme, issues with ortb2 as well as the docs on the website reop

**Comment by aleksatr** [comment]
@htang555 your tests are failing:

```
✖ 1 test failed

FAILED TESTS:
  DatablocksAdapter
    buildRequests
      ✖ Should be a valid openRTB request
        HeadlessChrome 89.0.4389 (Linux 0.0.0)
      AssertionError: expected { Object (id, tagid, ...) } to have keys 'banner', 'id', 'secure', 'tagid', 'placement_id', 'insights', and 'floor'

      + expected - actual

       [
         "banner"
         "floor"
         "id"
      -  "ortb2"
      +  "insights"
         "placement_id"
         "secure"
         "tagid"
       ]
      
    at webpack:///test/spec/modules/datablocksBidAdapter_spec.js:476:35 <- test/test_index.js:223732:81
    at Array.forEach (<anonymous>)
    at Context.<anonymous> (webpack:///test/spec/modules/datablocksBidAdapter_spec.js:472:12 <- test/test_index.js:223727:12)
```

**Comment by jmayor** [comment]
Should be good to go now

**Comment by ChrisHuie** [comment]
@htang555 can you please resubmit this pr 🙏 Had to revert it since it appears it slipped through some of our testing and we need to probably still fix a few things like linting

**Comment by htang555** [comment]
@ChrisHuie Created another PR here:

https://github.com/prebid/Prebid.js/pull/6765

**Pull request merged**